velox/exec/tests/utils/TestIndexStorageConnector.h

```diff
@@ -0,0 +1,299 @@
+    return true;
+  }
+
+  folly::dynamic serialize() const override {
+    folly::dynamic obj = folly::dynamic::object;
+    obj["name"] = name();
+    obj["connectorId"] = connectorId();
+    obj["asyncLookup"] = asyncLookup_;
+    return obj;
+  }
+
+  static std::shared_ptr<TestIndexTableHandle> create(
+      const folly::dynamic& obj,
+      void* context) {
+    // NOTE: this is only for testing purpose so we don't support to serialize
+    // the table.
+    return std::make_shared<TestIndexTableHandle>(
+        obj["connectorId"].getString(), nullptr, obj["asyncLookup"].asBool());
+  }
+
+  static void registerSerDe() {
+    auto& registry = DeserializationWithContextRegistryForSharedPtr();
+    registry.Register("TestIndexTableHandle", create);
+  }
+
+  const std::shared_ptr<TestIndexTable>& indexTable() const {
+    return indexTable_;
+  }
+
+  /// If true, we returns the lookup result asynchronously for testing purpose.
+  bool asyncLookup() const {
+    return asyncLookup_;
+  }
+
+ private:
+  const std::shared_ptr<TestIndexTable> indexTable_;
+  const bool asyncLookup_;
+};
+
+class TestIndexSource : public facebook::velox::connector::IndexSource,
+                        public std::enable_shared_from_this<TestIndexSource> {
+ public:
+  TestIndexSource(
+      const facebook::velox::RowTypePtr& inputType,
+      const facebook::velox::RowTypePtr& outputType,
+      const std::shared_ptr<facebook::velox::connector::ConnectorTableHandle>&
+          tableHandle,
+      facebook::velox::connector::ConnectorQueryCtx* connectorQueryCtx,
+      folly::Executor* executor);
+
+  std::unique_ptr<LookupResultIterator> lookup(
+      const LookupRequest& request) override;
+
+  std::unordered_map<std::string, facebook::velox::RuntimeCounter>
+  runtimeStats() override {
+    // TODO: add runtime stats.
+    return {};
+  }
+
+  class ResultIterator : public LookupResultIterator {
+   public:
+    ResultIterator(
+        std::shared_ptr<TestIndexSource> source,
+        const LookupRequest& request,
+        std::unique_ptr<HashLookup> lookupResult,
+        folly::Executor* executor);
+
+    std::optional<std::unique_ptr<LookupResult>> next(
+        vector_size_t size,
+        ContinueFuture& future) override;
+
+   private:
+#if 0
+    // Initializes the buffer used to storage row pointers in index table as the
+    // lookup matched result.
+    void initRowPointersBuffer(
+        BufferPtr& buffer,
+        char**& rawBuffer,
+        vector_size_t size);
+
+    // Initializes the buffer used to store the input row indices in the lookup
+    // request.
+    void initRowIndicesBuffer(
+        BufferPtr& buffer,
+        vector_size_t*& rawBuffer,
+        vector_size_t size);
+#endif
+
+    // Initializes the buffer used to storage row pointers or indices for output
+    // match result processing.
+    template <typename T>
+    void initBuffer(BufferPtr& buffer, T*& rawBuffer, vector_size_t size) {
```

Comment:
nit: size is an input parameter, buffer and rawBuffer as in/out parameters; if so, move 'size' before the other parameters